libp2p/peerstore.nim

```diff
@@ -53,6 +54,8 @@ type
 
     agentBook*: PeerBook[string]
     protoVersionBook*: PeerBook[string]
+
+    signedPeerRecordBook*: PeerBook[Envelope]
```

Comment:
Maybe it would make sense to create a specialized PeerBook that check the seqNo before storing?
Ofc we can trust the peer during identify, but if we get SPR from elsewhere it would be nice to just check here, wdyt ?

Reply:
Ah, do you mean to just check that the `seq` no increases monotonically? I didn't initially think it necessary for info we receive via `identify`, as we implicitly trust that information as being part of the negotiation.

I understand it may be necessary if we get the SPR "from elsewhere". In this case, wouldn't we want to check the signature as well? Also, I imagined that it's only in `PRUNE` that we'll receive SPR from another source, in which case it will be up to the higher layer application to verify the records and decide whether to connect or not? I may be missing something obvious here. :)

Reply:
I though a bit more about this, I think a proc in the PeerStore that checks the seqNo and update the relevant books when we get a new SPR would be best. I can add it as part of gspx